src/Text/Libyaml.hs

```diff
@@ -229,12 +239,54 @@ foreign import ccall unsafe "get_scalar_anchor"
 foreign import ccall unsafe "get_sequence_start_anchor"
     c_get_sequence_start_anchor :: EventRaw -> IO CString
 
+foreign import ccall unsafe "get_sequence_start_style"
+    c_get_sequence_start_style :: EventRaw -> IO CInt
+
+foreign import ccall unsafe "get_sequence_start_tag"
+    c_get_sequence_start_tag :: EventRaw -> IO (Ptr CUChar)
+
+foreign import ccall unsafe "get_sequence_start_tag_len"
+    c_get_sequence_start_tag_len :: EventRaw -> IO CULong
+
 foreign import ccall unsafe "get_mapping_start_anchor"
     c_get_mapping_start_anchor :: EventRaw -> IO CString
 
+foreign import ccall unsafe "get_mapping_start_style"
+    c_get_mapping_start_style :: EventRaw -> IO CInt
+
+foreign import ccall unsafe "get_mapping_start_tag"
+    c_get_mapping_start_tag :: EventRaw -> IO (Ptr CUChar)
+
+foreign import ccall unsafe "get_mapping_start_tag_len"
+    c_get_mapping_start_tag_len :: EventRaw -> IO CULong
+
 foreign import ccall unsafe "get_alias_anchor"
     c_get_alias_anchor :: EventRaw -> IO CString
 
+readAnchor :: (EventRaw -> IO CString) -> (EventRaw -> IO Anchor)
+readAnchor getAnchor er = do
+  yanchor <- getAnchor er
+  if yanchor == nullPtr
+    then return Nothing
+    else Just <$> peekCString yanchor
+
+readStyle :: (Enum a) => (EventRaw -> IO CInt) -> (EventRaw -> IO a)
+readStyle getStyle er = do
+  ystyle <- getStyle er
+  return $ toEnum $ fromEnum ystyle
+
+readTag ::
+     (EventRaw -> IO (Ptr CUChar)) -> (EventRaw -> IO CULong) -> (EventRaw -> IO Tag)
+readTag getTag getTagLen er = do
+  ytag <- getTag er
+  ytag_len <- getTagLen er
+  let ytag' = castPtr ytag
+  let ytag_len' = fromEnum ytag_len
+  bsToTag <$>
+    if ytag_len' == 0
+      then return Data.ByteString.empty
+      else packCStringLen (ytag', ytag_len')
```

Comment:
Given that under the surface the C code is just using `strlen` anyway to look for a null-terminating byte, is there a reason to use `packCStringLen` instead of `packCString`? It seems like the latter would allow us to simplify the changeset a bit.